src/plugins/initiatives/discourse.js

```diff
@@ -0,0 +1,72 @@
+// @flow
+
+import type {Topic, Post} from "../discourse/fetch";
+import type {Initiative, URL} from "./initiative";
+import {parseCookedHtml} from "./htmlTemplate";
+import {topicAddress} from "../discourse/address";
+
+/**
+ * Uses data from a Discourse Topic to create an Initiative representation.
+ *
+ * The Post should be the opening post of the Topic.
+ * The Post body should adhere to the `parseCookedHtml` expected template.
+ */
+export function initiativeFromDiscourseTracker(
+  serverUrl: string,
+  topic: Topic,
+  openingPost: Post
+): Initiative {
+  if (serverUrl.endsWith("/")) {
+    throw new Error("serverUrl shouldn't end with trailing slash");
+  }
+
+  const {title} = topic;
+  const {timestampMs} = openingPost;
+  try {
+    if (openingPost.topicId !== topic.id) {
+      throw new Error(`Post ${openingPost.id} is from a different topic`);
+    }
+
+    if (openingPost.indexWithinTopic !== 1) {
+      throw new Error(
+        `Post ${openingPost.id} is not the first post in the topic`
+      );
+    }
+
+    const tracker = topicAddress(serverUrl, topic.id);
+    const partial = parseCookedHtml(openingPost.cooked);
+    return {
+      title,
+      tracker,
+      timestampMs,
+      completed: partial.completed,
+      dependencies: absoluteURLs(serverUrl, partial.dependencies),
+      references: absoluteURLs(serverUrl, partial.references),
+      contributions: absoluteURLs(serverUrl, partial.contributions),
+      champions: absoluteURLs(serverUrl, partial.champions),
+    };
+  } catch (e) {
+    // To make solving issues easier, add which initiative topic caused the problem.
+    e.message = `${e.message} for initiative topic "${title}" (${topic.id})`;
```

Comment:
i think putting the url of the topic is even more helpful than the title and id

Reply:
I did find myself wanting to open these up from the terminal when I went back to edit wikis that triggered errors. Would be good to have urls.

Reply:
Now using URLs